src/addons/shape.js

```diff
@@ -0,0 +1,67 @@
+
+const shape = (G) => {
+
+  const simplifyDPStep = (points, first, last, sqTolerance, simplified) => {
+    const getSqSegDist = (p, p1, p2) => {
+      const getSqDiff = (a, b) => (a - b) * (a - b);
+      const getSqDist = (p1, p2) => getSqDiff(p1.x, p2.x) + getSqDiff(p1.y, p2.y);
+
+      if (p1.x === p2.x && p1.y === p2.y) return getSqDist(p, p1);
+
+      const t = ((p.x - p1.x) * (p2.x - p1.x) + (p.y - p1.y) * (p2.y - p1.y)) / getSqDist(p2, p1);
+      if (t > 1) return getSqDist(p, p2);
+      else if (t > 0) return getSqDist(p, { x: p1.x + (p2.x - p1.x) * t, y: p1.y + (p2.y - p1.y) * t });
+      else return getSqDist(p, p1);
+    }
+
+    let maxSqDist = sqTolerance, index;
+
+    for (let i = first + 1; i < last; i++) {
+      const sqDist = getSqSegDist(points[i], points[first], points[last]);
+      if (sqDist > maxSqDist) {
+        index = i;
+        maxSqDist = sqDist;
+      }
+    }
+
+    if (maxSqDist > sqTolerance) {
+      if (index - first > 1) simplifyDPStep(points, first, index, sqTolerance, simplified);
+      simplified.push(points[index]);
+      if (last - index > 1) simplifyDPStep(points, index, last, sqTolerance, simplified);
+    }
+  }
+
+  const simplifyDouglasPeucker = (points, sqTolerance) => {
```

Comment:
Can you add a link to the algorithm? Like an URL or something?

Can you also comment on what sqTolerance is and how to interpret/use it?